src/ui/utils/testRuns.ts

```diff
@@ -0,0 +1,72 @@
+import orderBy from "lodash/orderBy";
+
+import { Recording } from "shared/graphql/types";
+
+function testPassed(recording: Recording) {
+  return recording.metadata?.test?.result === "passed";
+}
+
+function testFailed(recording: Recording) {
+  return (
+    recording.metadata?.test?.result &&
+    ["failed", "timedOut"].includes(recording.metadata?.test?.result)
+  );
+}
+export function groupRecordings(recordings: Recording[] | undefined) {
+  const passedRecordings: Recording[] = [];
+  const failedRecordings: Recording[] = [];
+  const flakyRecordings: Recording[] = [];
+  const sortedRecordings = orderBy(recordings, "date", "desc");
+  const flakySet = new Set<string>();
+
+  const fileCounts = sortedRecordings.reduce((acc, recording) => {
+    const file = recording.metadata?.test?.file;
+    if (!file) {
+      return acc;
+    }
+    if (!acc[file]) {
+      acc[file] = 0;
+    }
+    acc[file]++;
+    return acc;
+  }, {} as Record<string, number>);
+
+  for (const recording of sortedRecordings) {
+    const file = recording.metadata?.test?.file;
+
+    if (!file) {
+      continue;
+    }
+    if (fileCounts[file] > 1) {
+      const recordings = sortedRecordings.filter(r => r.metadata?.test?.file === file);
+      const passed = recordings.filter(r => testPassed(r));
+      const failed = recordings.filter(r => testFailed(r));
+
+      // We currently only show the first failed recording.
+      // What we'd eally want to do is show all the recordings as a group
+      if (passed.length > 0 && failed.length > 0) {
+        flakyRecordings.push(recording);
+        flakySet.add(file);
+      }
```

Comment:
Maybe this is the place we'd want to increment either the flaky, passed, or failed counts?

```js
if (passed.length > 0 && failed.length > 0) {
  flakyRecordings.push(recording);
  flakySet.add(file);
} else if (passed.length > 0) {
  passedSet.add(file);
} else {
  failedSet.add(file);
}
```

Reply:
This is a bit subtle, but for failed and passed i do want to use `.length` because it's fine to have multiple tests in different groups. Whereas for flakes, those are the same tests that are retried

Reply:
Hmm, I hear what you're saying, but I think it's actually pretty confusing UX to count a test as both "flaky" and "failed" (at least in the summary view and/or for how you group it).

Reply:
In the screenshot of this change, for example:
<img width="1317" alt="Screen Shot 2023-05-26 at 8 16 11 AM" src="https://github.com/replayio/devtools/assets/29597/d0b500da-ceac-4d31-bdb6-ad43a12e9c48">

The summary row shows a failed status (presumably because `failed.length > 0`) but it shouldn't. It should show a flaky color at worst.